packages/contracts/test/framework/plugin/plugin-setup-processor.ts

```diff
@@ -577,157 +789,253 @@ describe.skip('Plugin Setup Processor', function () {
+              psp,
+              targetDao.address,
+              proxy,
+              // @ts-ignore
+              modifiedPluginRepoPointer,
+              helpersUV1,
+              EMPTY_DATA
+            )
+          )
+            .to.be.revertedWithCustomError(psp, 'InvalidAppliedSetupId')
+            .withArgs(currentAppliedSetupId, appliedSetupId);
+        }
+      });
+
+      it('reverts if plugin uninstallation with the same setup is already prepared', async () => {
+        const {preparedSetupId} = await prepareUninstallation(
           psp,
           targetDao.address,
           proxy,
-          setupUV1.address,
-          repoU.address,
+          pluginRepoPointer,
           helpersUV1,
           EMPTY_DATA
         );
 
-        // prepare second uninstallation
         await expect(
           prepareUninstallation(
             psp,
             targetDao.address,
             proxy,
-            setupUV1.address,
-            repoU.address,
+            pluginRepoPointer,
             helpersUV1,
             EMPTY_DATA
           )
-        ).to.be.revertedWithCustomError(psp, 'SetupAlreadyPrepared');
+        )
+          .to.be.revertedWithCustomError(psp, 'SetupAlreadyPrepared')
+          .withArgs(preparedSetupId);
       });
 
-      it('prepares an UUPS upgradeable plugin uninstallation', async () => {
-        let returnedPlugin;
-        let returnedHelpersV1;
-        let uninstallPermissionsV1;
-        ({
-          returnedPluginAddress: returnedPlugin,
-          returnedHelpers: returnedHelpersV1,
-          permissions: uninstallPermissionsV1,
-        } = await prepareUninstallation(
+      it('reverts if the plugin was uninstalled and tries to prepare uninstallation for it', async () => {
+        // make sure that prepare uninstall doesn't revert before applying uninstall.
+        await expect(
+          psp.callStatic.prepareUninstallation(
+            targetDao.address,
+            createPrepareUninstallationParams(
+              proxy,
+              pluginRepoPointer,
+              helpersUV1,
+              EMPTY_DATA
+            )
+          )
+        ).not.to.be.reverted;
+
+        await uninstallPlugin(
           psp,
           targetDao.address,
           proxy,
-          setupUV1.address,
-          repoU.address,
           helpersUV1,
+          pluginRepoPointer,
           EMPTY_DATA
-        ));
+        );
 
-        expect(returnedPlugin).to.not.equal(AddressZero);
-        expect(returnedHelpersV1).to.deep.equal(mockHelpers(1));
-        expect(uninstallPermissionsV1).to.deep.equal(
-          mockPermissionsOperations(0, 1, Operation.Revoke).map(perm =>
-            Object.values(perm)
+        await expect(
+          psp.prepareUninstallation(
+            targetDao.address,
+            createPrepareUninstallationParams(
+              proxy,
+              pluginRepoPointer,
+              helpersUV1,
+              EMPTY_DATA
+            )
```

Comment:
check that it reverts with `InvalidAppliedSetupId` so we know that we landed in the right code path